Library/OcDebugLogLib/OcLog.c

```diff
@@ -140,18 +146,138 @@ GetLogPath (
   return LogPath;
 }
 
+STATIC
 EFI_STATUS
-EFIAPI
-OcLogAddEntry  (
-  IN OC_LOG_PROTOCOL    *OcLog,
-  IN UINTN              ErrorLevel,
-  IN CONST CHAR8        *FormatString,
-  IN VA_LIST            Marker
+GetLogPrefix (
+  IN   CONST CHAR8  *FormatString,
+  OUT  CHAR8        *Prefix
   )
 {
-  EFI_STATUS                  Status;
+  UINTN    MaxLength;
+  UINTN    Index;
+  CHAR8    Curr;
 
-  OC_LOG_PRIVATE_DATA         *Private;
+  ASSERT (FormatString != NULL);
+  ASSERT (Prefix != NULL);
+
+  //
+  // If FormatString just starts with colon, it must be illegal.
+  //
+  if (*FormatString == ':') {
+    return EFI_NOT_FOUND;
+  }
+
+  MaxLength = MIN (AsciiStrLen (FormatString), OC_LOG_PREFIX_CHAR_MAX);
+  for (Index = 1; Index < MaxLength; ++Index) {
+    Curr = FormatString[Index];
+
+    //
+    // Match the first occurrence of colon.
+    //
+    if (Curr == ':') {
+      break;
+    }
+
+    //
+    // Except for colon, a valid prefix must be either 0-9, or uppercase letter.
+    //
+    if (!(IsAsciiNumber (Curr) || (Curr >= 'A' && Curr <= 'Z'))) {
+      return EFI_NOT_FOUND;
+    }
+  }
+
+  //
+  // If Index went through the end, then ':' was not found.
+  //
+  if (Index == MaxLength) {
+    return EFI_NOT_FOUND;
+  }
+
+  CopyMem (Prefix, FormatString, Index);
+  Prefix[Index] = '\0';
+  return EFI_SUCCESS;
+}
+
+STATIC
+BOOLEAN
+IsPrefixFiltered (
+  IN   CONST CHAR8          *FormatString,
+  IN   CONST OC_FLEX_ARRAY  *FlexFilters,
```

Comment:
Should be optional if it allows NULL.

Reply:
https://github.com/acidanthera/OpenCorePkg/commit/2f5c720bf2aef42aa233c7dd222167aa664d8440